### PR TITLE
Particles - Fix macro

### PR DESCRIPTION
**Pull request by TyroneMF**
**When merged this pull request will:**
- Fix (#8183)

### CODE REVIEW

File: optionals/particles/script_component.hpp [hunk 1]
```diff
@@ -51,5 +51,5 @@
 
 #define CLOUDLET_UNDERWATER(color) class SmokeShell##color##UW; \
 class SmokeShell##color##2UW: SmokeShell##color##UW { \
-    postEffects = QUOTE(ACE_SmokeAfterWater##color); \
+    postEffects = QUOTE(DOUBLES(ACE_SmokeAfterWater,color)); \
```
Comment (commy2) [inline optionals/particles/script_component.hpp:54]:
```suggestion
    postEffects = 'ACE_SmokeAfterWater##color'; \
```

Comment (TyroneMF) [reply optionals/particles/script_component.hpp:54]: Tried this, now resolves in-game as
`type = "ACE_SmokeAfterWaterWhite";` & `postEffects = "'ACE_SmokeAfterWaterWhite'";`

Same error pops when testing though.
> 20:53:32 Warning Message: No entry 'bin\config.bin.'ACE_SmokeAfterWaterWhite''.

Comment (commy2) [reply optionals/particles/script_component.hpp:54]:
```suggestion
    postEffects = ACE_SmokeAfterWater##color; \
```

**Comment by jonpas** [comment]
Can you try:
- `'ACE_SmokeAfterWater####color'`
- Go back to [this](https://github.com/acemod/ACE3/commit/326604ae9fa2f88a851957bc966ec9903960e57e) and see if it works with both pboProject and HEMTT.

File: optionals/particles/script_component.hpp [hunk 2]
```diff
@@ -30,26 +30,26 @@
...
     class SmokeAfterWater: SmokeAfterWater { \
-        type = QUOTE(ACE_SmokeAfterWater##color); \
+        type = ACE_SmokeAfterWater##color; \
     }; \
 }
 
 #define CLOUDLET_UNDERWATER(color) class SmokeShell##color##UW; \
 class SmokeShell##color##2UW: SmokeShell##color##UW { \
-    postEffects = QUOTE(DOUBLES(ACE_SmokeAfterWater,color)); \
+    postEffects = ACE_SmokeAfterWater,color; \
```
Comment (commy2) [inline optionals/particles/script_component.hpp:54]: no

**Comment by TyroneMF** [comment]
> 
> 
> Can you try:
> 
>     * `'ACE_SmokeAfterWater####color'`
> 
>     * Go back to [this](https://github.com/acemod/ACE3/commit/326604ae9fa2f88a851957bc966ec9903960e57e) and see if it works with both pboProject and HEMTT.

Works & builds with both.

**Comment by jonpas** [comment]
Which one?

**Comment by TyroneMF** [comment]
Both ways build.
Only this one works:
`#define MERGE(var1,var2) var1####var2` 
`type = QUOTE(ACE_SmokeAfterWater##color); \`

using `####` ends in double quotes.

**Comment by jonpas** [comment]
Apply that then, if it works with both it works.

**Comment by TyroneMF** [comment]
It's done, it builds, it works and it doesn't error.

**Review by jonpas — APPROVED** [review]

**Review by veteran29 — APPROVED** [review]

**Pull request merged**